packages/ruleset/src/functions/pagination-style.js

```diff
@@ -0,0 +1,363 @@
+    }
+  }
+
+  // Check #7: The response body must contain an array property whose name matches the final path segment.
+  // Reference: https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-collections-overview#response-format
+  const pathSeg = pathStr.split('/').pop();
+  const resourcesProp = responseSchema.properties[pathSeg];
+  if (!resourcesProp || resourcesProp.type !== 'array') {
+    results.push({
+      message:
+        'A paginated list operation must include an array property whose name matches the final segment of the path',
+      path: responseSchemaPath
+    });
+  }
+
+  // Check #8: If the response body contains a "total_count" property, it must be type integer and required.
+  // References:
+  // - https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-pagination#additional-response-fields-with-token
+  // - https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-pagination#additional-response-fields-with-offset
+  const tcProp = responseSchema.properties.total_count;
+  if (tcProp) {
+    if (
+      tcProp.type !== 'integer' ||
+      !responseSchema.required ||
+      responseSchema.required.indexOf('total_count') === -1
+    ) {
+      results.push({
+        message:
+          'The "total_count" property in the response body of a paginated list operation must be of type integer and required',
+        path: [...responseSchemaPath, 'properties', 'total_count']
+      });
+    }
+  }
+
+  // Checks 9-12 below will verify that the response schema contains the
+  // "first", "last", "previous", and "next" properties which are supposed to
+  // provide links (in the form of a request url string) to the first/last/previous/next result pages.
+  // Each of these properties should be an object that contains an "href" string property.
+  // Reference: https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-pagination#pagination-links
+
+  // Check #9: The response body should contain a "first" property which links to the first page of results.
+  results.push(...checkPageLink(responseSchemaPath, responseSchema, 'first'));
+
+  // Check #10: The response body should contain a "last" property which links to the last page of results.
+  results.push(...checkPageLink(responseSchemaPath, responseSchema, 'last'));
+
+  // Check #11: The response body should contain a "previous" property which links to the previous page of results.
+  results.push(
+    ...checkPageLink(responseSchemaPath, responseSchema, 'previous')
+  );
+
+  // Check #12: The response body should contain a "next" property which links to the next page of results.
+  results.push(...checkPageLink(responseSchemaPath, responseSchema, 'next'));
+
+  if (results.length > 0) {
+    debug('Results: ' + JSON.stringify(results, null, 2));
+  } else {
+    debug('PASSED');
+  }
+
+  return results;
+}
+
+// Verify that the '<name>' property exists within 'responseSchema'
+// and represents a valid "page link" value (an object with an "href" field).
```

Comment:
Should we also check for the page token to be reflected? (Maybe a future issue.)

Reply:
My understanding from the API handbook is that the page token query param should be returned by the server as part of the URL string within the "href" field.   If that's the case, then I don't know what we could possibly do in the validator since we are not validating the actual responses coming back from the server (we're validating the schema within the API definition).